Dev MCP Documentation (#49105)
quarkusio/quarkus: docs/src/main/asciidoc/dev-mcp.adoc

////
This guide is maintained in the main Quarkus repository
and pull requests should be submitted there:
https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
////
= Dev MCP
include::_attributes.adoc[]
:categories: writing-extensions
:summary: Learn more about Dev MCP
:topics: dev-mcp,dev-ui

[IMPORTANT]
.Experimental
====
This feature is currently experimental. To enable it, set `quarkus.dev-mcp.enabled=true` in your application properties.
====

== Overview

When you run a Quarkus application in dev mode, you can expose a Model Context Protocol (MCP) server. 
The Dev MCP server surfaces some of the same JSON‑RPC services and build‑time metadata that the Dev UI uses, but via the MCP protocol. 

> It presents this data as *MCP tools* (methods you can call) and *MCP resources* (data you can inspect).

Author: You mean before the IMPORTANT note?

> It presents this data as *MCP tools* (methods you can call) and *MCP resources* (data you can inspect).

Reviewer: No no, I mean it should be the first paragraph of Overview

////
This guide is maintained in the main Quarkus repository
and pull requests should be submitted there:
https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
////
= Dev MCP
include::_attributes.adoc[]
:categories: writing-extensions
:summary: Learn more about Dev MCP
:topics: dev-mcp,dev-ui

[IMPORTANT]
.Experimental
====
This feature is currently experimental. To enable it, set `quarkus.dev-mcp.enabled=true` in your application properties.
====

== Overview

> When you run a Quarkus application in dev mode, you can expose a Model Context Protocol (MCP) server. 
> The Dev MCP server surfaces some of the same JSON‑RPC services and build‑time metadata that the Dev UI uses, but via the MCP protocol. 
> It presents this data as *MCP tools* (methods you can call) and *MCP resources* (data you can inspect).

Reviewer: ```suggestion
When you run a Quarkus application in dev mode, you can expose a Model Context Protocol (MCP) server. 
It presents the methods used by the DevUI *MCP tools* (methods you can call) and the data exposed by *MCP resources*. 

```
